### ISSUE TITLE
debug_traceCall chokes (never returns)

### ISSUE THREAD

**Issue by drortirosh** [paradigmxyz/reth#4179]
### Describe the bug

trying to test the compatibiltiy of debug_traceCall with a js "tracer", I found out that the server chokes and doesn't return at all when the following command is called:

```
curl -s -d '{"method":"debug_traceCall","params":[{},"latest",{"tracer":"{ fault: function(log, db) { }, result: function(ctx, db) { return {}; }, step: function(log, db) { } }"}],"id":1234,"jsonrpc":"2.0"}' -H content-type:application/json http://localhost:8545
```

The same ch
If the "tracer" parameter is removed (or even renamed, so the parser would ignore it), it does attempt a traceCall (but of course, without a tracer)

The same problem persist even if a full transaction (with from/to/data) are provided

(I call this a "bug" and not "vulnerability", since the debug API shouldn't be open to the public in any case, but of course, it blocks a working threads and can easily block the server completely.)

### Steps to reproduce

1. make install
2. reth node -vvvvvv --dev --http  --http.api eth,debug
3. curl -s -d '{"method":"debug_traceCall","params":[{},"latest",{"tracer":""}],"id":1234,"jsonrpc":"2.0"}' -H content-type:application/json http://localhost:8545

### Platform(s)

Mac (Apple Silicon)

### What version/commit are you on?

reth 0.1.0-alpha.6 (84625e4a)

### If you've built Reth from source, provide the full command you used

make install

### Code of Conduct

- [X] I agree to follow the Code of Conduct

**Comment by mattsse** [comment]
thanks for flagging, fixed in #4180

> curl -s -d '{"method":"debug_traceCall","params":[{},"latest",{"tracer":"{ fault: function(log, db) { }, result: function(ctx, db) { return {}; }, step: function(log, db) { } }"}],"id":1234,"jsonrpc":"2.0"}' -H content-type:application/json http://localhost:8545

> {"jsonrpc":"2.0","result":{},"id":1234}

I also found that in the case of invalid tracer object we currently return

```
{
    "jsonrpc": "2.0",
    "error": {
        "code": -32602,
        "message": "Failed to eval js code: SyntaxError: unexpected token ')', empty parenthesized expression at line 1, col 2"
    },
    "id": 1
}
```

geth and erigon return different errors here

"Error: -32602, Tracer object with 'tracer' param is required for debug_traceCall

tracer not found

which we could also adapt

**Comment by mattsse** [comment]
also the js tracer impl lacks same test coverage, so likely that there are still a few additional bugs
